src/common/src/config.rs

```diff
@@ -2420,19 +2424,56 @@ pub struct StorageMemoryConfig {
     pub meta_cache_capacity_mb: usize,
     pub meta_cache_shard_num: usize,
     pub shared_buffer_capacity_mb: usize,
-    pub compactor_memory_limit_mb: usize,
     pub prefetch_buffer_capacity_mb: usize,
     pub block_cache_eviction_config: EvictionConfig,
     pub meta_cache_eviction_config: EvictionConfig,
-    pub block_file_cache_flush_buffer_threshold_mb: usize,
+    pub data_file_cache_flush_buffer_threshold_mb: usize,
     pub meta_file_cache_flush_buffer_threshold_mb: usize,
 }
 
+pub struct CompactorMemoryConfig {
+    pub compactor_meta_cache_capacity_mb: usize,
+    pub compactor_memory_limit_mb: usize,
+}
+
 pub const MAX_META_CACHE_SHARD_BITS: usize = 4;
 pub const MIN_BUFFER_SIZE_PER_SHARD: usize = 256;
 pub const MAX_BLOCK_CACHE_SHARD_BITS: usize = 6; // It means that there will be 64 shards lru-cache to avoid lock conflict.
 
-pub fn extract_storage_memory_config(s: &RwConfig) -> StorageMemoryConfig {
+pub const COMPACTOR_MEMORY_PROPORTION: f64 = 0.1;
+
+// calculate the compactor meta cache capacity based on capacity and memory proportion
+// compactor_meta_cache_capacity_mb = min(MIN_COMPACTOR_META_CACHE_CAPACITY_MB, memory * MAX_COMPACTOR_META_CACHE_CAPACITY_PROPORTION)
```

Comment:
Is `memory` in the docs here referring to memory used by compactor or overall standalone process memory.

Reply:
Both compactor and standalone.

1. dedicated compactor:  extract_compactor_memory_config = 1.0
2. standalone compactor: extract_compactor_memory_config = COMPACTOR_MEMORY_PROPORTION = 0. 1

And it doesn't change the original memory ratios.